pages/api/dpu/pubsub.ts

```diff
@@ -2,48 +2,60 @@ import { NextApiRequest, NextApiResponse } from 'next';
 import { DbUser, getUserById } from '../../../utils/db/users';
+import rudderStackEvents from '../events';
 
 const pubsubHandler = async (req: NextApiRequest, res: NextApiResponse) => { // To be removed, only used for testing the functions
 	console.info("[pubsubHandler] pub sub setup info in db...");
 	const jsonBody = req.body;
 	if (!jsonBody.userId) {
 		console.error("[pubsubHandler] Invalid request body");
 		res.status(400).json({ "error": "Invalid request body" });
+		const eventProperties = { ...jsonBody, response_status: 400 };
+		rudderStackEvents.track(jsonBody.userId, "", 'dpu/pubsub', { type: 'create-topic', eventStatusFlag: 0, eventProperties });
 		return;
 	}
 	const userData: DbUser = await getUserById(jsonBody.userId);
 	if (!userData) {
 		console.error(`[pubsubHandler] cannot get userData`);
 		res.status(500).json({ "error": "Internal server error" });
+		const eventProperties = { ...jsonBody, response_status: 500 };
+		rudderStackEvents.track(jsonBody.userId, "", 'dpu/pubsub', { type: 'create-topic', eventStatusFlag: 0, eventProperties });
 		return;
 	}
 	if (!userData.topic_name) {
 		const generatedTopic = await createTopicName(jsonBody.userId);
 		if (!generatedTopic) {
 			console.error(`[pubsubHandler] error in creating topic name`);
 			res.status(500).json({ "error": "Internal server error" });
+			const eventProperties = { ...jsonBody, response_status: 500 };
+			rudderStackEvents.track(jsonBody.userId, "", 'dpu/pubsub', { type: 'create-topic', eventStatusFlag: 0, eventProperties });
 			return;
 		}
 		const gcloudTopic = await createTopicNameInGcloud(generatedTopic)
 		if (!gcloudTopic) {
 			console.error(`[pubsubHandler] error in creating topic in google cloud`);
 			res.status(500).json({ "error": "Internal server error" });
+			const eventProperties = { ...jsonBody, response_status: 500 };
+			rudderStackEvents.track(jsonBody.userId, "", 'dpu/pubsub', { type: 'create-topic', eventStatusFlag: 0, eventProperties });
 			return;
 		}
 		await saveTopicNameInUsersTable(jsonBody.userId, generatedTopic)
 			.then(() => {
 				console.info("[pubsubHandler] Topic saved in db");
-				res.status(200).json({ "installId": generatedTopic });
 			})
 			.catch((error) => {
 				console.error("[pubsubHandler] Unable to save topic name in db, ", error);
 				res.status(500).json({ "error": "Internal server error" });
+				const eventProperties = { ...jsonBody, response_status: 500 };
+				rudderStackEvents.track(jsonBody.userId, "", 'dpu/pubsub', { type: 'create-topic', eventStatusFlag: 0, eventProperties });
 			})
 		return;
 	}
 	const topicName = userData.topic_name;
 	console.info("[pubsubHandler] topic name created successfully and saved in db: ", topicName);
 	res.status(200).json({ "installId": topicName });
+	const eventProperties = { ...jsonBody, topicName, response_status: 200 };
+	rudderStackEvents.track(jsonBody.userId, "", 'dpu/pubsub', { type: 'create-topic', eventStatusFlag: 1, eventProperties });
```

Comment:
Is `create-topic` the best event name here?
It seems that in this handler, we are first checking if the topic name exists, and only creating the topic if it doesn't. 

A couple of suggestions for the name of this event: `get-pubsub-topic`, `get-user-pubsub-topic`, `get-user-topicname`, `get-topic` etc.

Also, I think we can still use the `create-topic` event name within the `(!userData.topicname)` block.